src/containers/IdeLoader.tsx

```diff
@@ -0,0 +1,190 @@
+/*
+ * Copyright (c) 2018-2020 Red Hat, Inc.
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *   Red Hat, Inc. - initial API and implementation
+ */
+
+import { AlertVariant } from '@patternfly/react-core';
+import React from 'react';
+import { connect, ConnectedProps } from 'react-redux';
+import { History } from 'history';
+import { RouteComponentProps } from 'react-router';
+import { timeout } from '../services/timeout';
+import { AppState } from '../store';
+import * as WorkspaceStore from '../store/Workspaces';
+import IdeLoaderPage from '../pages/IdeLoader';
+import { selectAllWorkspaces, selectWorkspaceById } from '../store/Workspaces/selectors';
+import { WorkspaceStatus } from '../services/workspaceStatus';
+
+type Props =
+  MappedProps
+  & { history: History }
+  & RouteComponentProps<{ namespace: string; workspaceName: string }>;
+
+type State = {
+  namespace?: string,
+  workspaceName?: string,
+  workspaceId?: string,
+  currentStep: number;
```

Comment:
WDYT of having an enum of IDE loading steps instead of numbers? I think it will be much more convenient to handle them as they'll be self-descriptive.

Reply:
Ok. I have fixed it.